framework/Models.py

```diff
@@ -492,75 +509,51 @@ def _inputToInternal(self,dataIN,full=False):
-    """
-      Method to update an input from outside
-      @ In, Input, list, list of inputs that needs to be updated
-      @ In, externalDict, dict, dictionary of new values that need to be added or updated
-      @ Out, inputOut, list, updated list of inputs
-    """
-    inputOut = Input
-    for key, value in externalDict.items():
-      inputOut[0][0][key] =  externalDict[key]
-      inputOut[1]["SampledVars"  ][key] =  externalDict[key]
-      inputOut[1]["SampledVarsPb"][key] =  1.0    #FIXME it is a mistake (Andrea). The SampledVarsPb for this variable should be transfred from outside
-      self._replaceVariablesNamesWithAliasSystem(inputOut[1]["SampledVars"  ],'input',False)
-      self._replaceVariablesNamesWithAliasSystem(inputOut[1]["SampledVarsPb"],'input',False)
-    return inputOut
-
-  def run(self,Input,jobHandler):
-    """
-      This method executes the model .
-      @ In,  Input, object, object contained the data to process. (inputToInternal output)
-      @ In,  jobHandler, JobHandler instance, the global job handler instance
-      @ Out, None
+    if 'SampledVars' in kwargs.keys() and len(self.alias['input'].keys()) != 0:
+      kwargs['SampledVars'] = sampledVars
+    return [(inputDict)],copy.deepcopy(kwargs)
+
+  def evaluateSample(self, myInput, samplerType, kwargs):
+    """
+        This will evaluate an individual sample on this model. Note, parameters
+        are needed by createNewInput and thus descriptions are copied from there.
+        @ In, myInput, list, the inputs (list) to start from to generate the new one
+        @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+        @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+        @ Out, returnValue, dict, This holds the output information of the evaluated sample.
     """
-    #this set of test is performed to avoid that if used in a single run we come in with the wrong input structure since the self.createNewInput is not called
+    Input = self.createNewInput(myInput, samplerType, **kwargs)
     inRun = self._manipulateInput(Input[0])
-
-    def lambdaReturnOut(inRun,prefix):
-      """
-        This method is the one is going to be submitted through the jobHandler
-        @ In, inRun, dict, the input
-        @ In, prefix, string, the string identifying this job
-        @ Out, lambdaReturnOut, dict, the return dictionary
-      """
-      return {'OutputPlaceHolder':np.atleast_1d(np.float(prefix))}
-
-    uniqueHandler = Input[1]['uniqueHandler'] if 'uniqueHandler' in Input[1].keys() else 'any'
-    jobHandler.addInternal((inRun,Input[1]['prefix']),lambdaReturnOut,str(Input[1]['prefix']),metadata=Input[1], modulesToImport = self.mods, uniqueHandler=uniqueHandler)
+    return (inRun,{'OutputPlaceHolder':np.atleast_1d(np.float(Input[1]['prefix']))})
```

Comment:
Ah yes! You are right, it is now a tuple. I will be sure to address this for each of the Model classes and the base.